registry/auth.go

```diff
@@ -39,93 +35,34 @@ func loginV1(authConfig *types.AuthConfig, registryEndpoint *Endpoint) (string,
 
 	loginAgainstOfficialIndex := serverAddress == IndexServer
 
-	// to avoid sending the server address to the server it should be removed before being marshaled
-	authCopy := *authConfig
-	authCopy.ServerAddress = ""
-
-	jsonBody, err := json.Marshal(authCopy)
-	if err != nil {
-		return "", fmt.Errorf("Config Error: %s", err)
-	}
-
-	// using `bytes.NewReader(jsonBody)` here causes the server to respond with a 411 status.
-	b := strings.NewReader(string(jsonBody))
-	resp1, err := registryEndpoint.client.Post(serverAddress+"users/", "application/json; charset=utf-8", b)
+	req, err := http.NewRequest("GET", serverAddress+"users/", nil)
+	req.SetBasicAuth(authConfig.Username, authConfig.Password)
+	resp, err := registryEndpoint.client.Do(req)
 	if err != nil {
-		return "", fmt.Errorf("Server Error: %s", err)
+		return "", err
 	}
-	defer resp1.Body.Close()
-	respStatusCode = resp1.StatusCode
-	respBody, err = ioutil.ReadAll(resp1.Body)
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return "", fmt.Errorf("Server Error: [%#v] %s", respStatusCode, err)
+		return "", err
 	}
-
-	if respStatusCode == 201 {
+	if resp.StatusCode == 200 {
```

Comment:
Nit: replace these status code numbers with `http.StatusOK`, `http.StatusUnauthorized`, and `http.StatusForbidden`